Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstanding.cs

```diff
@@ -0,0 +1,244 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using System.Collections;
+using HoloToolkit.Unity;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// The SpatialUnderstanding class controls the state and flow of the 
+    /// scanning process used in the understanding module. 
+    /// </summary>
+    [RequireComponent(typeof(SpatialUnderstandingSourceMesh))]
+    [RequireComponent(typeof(SpatialUnderstandingCustomMesh))]
+    public class SpatialUnderstanding : Singleton<SpatialUnderstanding>
+    {
+        // Consts
+        public const float ScanSearchDistance = 8.0f;
+
+        // Enums
+        public enum ScanStates
+        {
+            None,
+            ReadyToScan,
+            Scanning,
+            Finishing,
+            Done
+        }
+
+        // Config
+        [Tooltip("If set to false, scanning will only begin after RequestBeginScanning is called")]
+        public bool AutoBeginScanning = true;
+        [Tooltip("Update period used during the scanning process (typically faster than after scanning is completed)")]
+        public float UpdatePeriod_DuringScanning = 1.0f;
+        [Tooltip("Update period used after the scanning process is completed")]
+        public float UpdatePeriod_AfterScanning = 4.0f;
+
+        // Properties
+        /// <summary>
+        /// Switch used by the entire SpatialUnderstanding module to activate processing.
+        /// </summary>
+        public bool AllowSpatialUnderstanding
+        {
+            get
+            {
+#if UNITY_METRO && !UNITY_EDITOR
+                return true;
+#endif
+                return false;
+            }
+        }
+        /// <summary>
+        /// Reference to the SpatialUnderstandingDLL class (wraps the understanding dll functions).
+        /// </summary>
+        public SpatialUnderstandingDll UnderstandingDLL { get; private set; }
```

Comment:
Again, don't know if this is a NIT or not, but according to the [.Net Capitalization Conventions](https://msdn.microsoft.com/en-us/library/ms229043(v=vs.110).aspx), DLL should not be upper case. 

> The PascalCasing convention, used for all identifiers except parameter names, capitalizes the first character of each word (including acronyms over two letters in length), as shown in the following examples:
> 
> PropertyDescriptor
> HtmlTag
> 
> A special case is made for two-letter acronyms in which both letters are capitalized, as shown in the following identifier:
> 
> IOStream